app/scripts/lib/transaction/hooks/delegation-7702-publish.ts

```diff
@@ -106,37 +106,58 @@ export class Delegation7702PublishHook {
     const { delegationAddress, upgradeContractAddress } =
       atomicBatchChainSupport;
 
-    if (!selectedGasFeeToken || !gasFeeTokens?.length) {
+    const isGasless7702Tx = await this.#isBridgeTxGasless7702(transactionMeta);
+
+    if ((!selectedGasFeeToken || !gasFeeTokens?.length) && !isGasless7702Tx) {
       log('Skipping as no selected gas fee token');
       return EMPTY_RESULT;
     }
 
-    const gasFeeToken = gasFeeTokens.find(
-      (token) =>
-        token.tokenAddress.toLowerCase() === selectedGasFeeToken.toLowerCase(),
-    );
+    const gasFeeToken = isGasless7702Tx
+      ? undefined
+      : gasFeeTokens?.find(
+          (token) =>
+            token.tokenAddress.toLowerCase() ===
+            selectedGasFeeToken?.toLowerCase(),
+        );
 
-    if (!gasFeeToken) {
+    if (!gasFeeToken && !isGasless7702Tx) {
       throw new Error('Selected gas fee token not found');
     }
 
-    const delegation = await this.#buildDelegation(
+    const delegationEnvironment = getDeleGatorEnvironment(
+      Number(hexToDecimal(transactionMeta.chainId)),
+    );
+    const delegationManagerAddress = delegationEnvironment.DelegationManager;
+    const includeTransfer = !isGasless7702Tx;
+    const delegation = this.#buildUnsignedDelegation(
+      delegationEnvironment,
       transactionMeta,
       gasFeeToken,
+      includeTransfer,
     );
 
-    const executions = this.#buildExecutions(transactionMeta, gasFeeToken);
+    log('Signing delegation');
 
-    const transactionData = encodeRedeemDelegations(
-      [[delegation]],
-      [ExecutionMode.BATCH_DEFAULT_MODE],
-      [executions],
-    );
+    const delegationSignature = (await this.#messenger.call(
+      'DelegationController:signDelegation',
+      {
+        chainId,
+        delegation,
+      },
+    )) as Hex;
+
+    log('Delegation signature', delegationSignature);
+
+    const transactionData = generateCalldata({
+      transaction: txParams,
+      delegation: { ...delegation, signature: delegationSignature },
+    });
 
     const relayRequest: RelaySubmitRequest = {
       chainId,
       data: transactionData,
-      to: process.env.DELEGATION_MANAGER_ADDRESS as Hex,
+      to: delegationManagerAddress,
```

Comment:
Can we also remove the ENVs from `builds.yml`?

Reply:
The env is used by `signDelegation` from `app/scripts/lib/transaction/delegation.ts` which is used in some tests with other utilities such as `encodeRedeemDelegations` that are still used. I will keep it for now.